### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by alina-kho**
Comments and suggestions are highly appreciated :) Thanks!

### CODE REVIEW

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+router.get("/", async (request, response) => {
+  try {
+    const supportedParams = ["maxPrice", "limit", "title", "createdAfter"]; //will be needed for query validation
+    const maxPrice = parseInt(request.query.maxPrice); //converting to int
+    const limit = parseInt(request.query.limit); //converting to int
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+
+    //Checking if there are any queries
+    if (Object.keys(request.query) != 0) {
```
Comment (jslambda) [inline nodejs/week2/meal-sharing/src/backend/api/meals-router.js]: Object.keys never returns 0 so the branch's condition is always true regardless of the query parameters.
Can you fix this?

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    const supportedParams = ["maxPrice", "limit", "title", "createdAfter"]; //will be needed for query validation
+    const maxPrice = parseInt(request.query.maxPrice); //converting to int
+    const limit = parseInt(request.query.limit); //converting to int
```
Comment (jslambda) [inline nodejs/week2/meal-sharing/src/backend/api/meals-router.js]: What would happen if price or limit parameters are not integers?
Can you check for these cases?

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      } else if (limit) {
+        response.send(meals.slice(0, limit));
+      } else if (!supportedParams.includes(request.query)) {
+        response.send(400, "This query parameter is not supported");
+      }
+    } else {
+      response.send(meals);
+    }
+  } catch (error) {
+    throw error;
```
Comment (jslambda) [inline nodejs/week2/meal-sharing/src/backend/api/meals-router.js:70]: What would happen if there are more than one valid parameter in the request? Can you modify your implementation to take all of them into account?

File: nodejs/week2/meal-sharing/src/backend/api/reservations-router.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    throw error;
+  }
+});
+
+router.get("/:id", async (request, response) => {
+  try {
+    if (parseInt(request.params.id)) {
+      response.send(
+        200,
+        reservations.find(
```
Comment (jslambda) [inline nodejs/week2/meal-sharing/src/backend/api/reservations-router.js:19]: Nice to use find here 👍

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js (hunk 1 above)
Comment (alina-kho) [reply nodejs/week2/meal-sharing/src/backend/api/meals-router.js]: Sure! It should be Object.keys(request.query).length I guess? Lost this "length" part while modifying the code several times.

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js (hunk 2 above)
Comment (alina-kho) [reply nodejs/week2/meal-sharing/src/backend/api/meals-router.js]: will do it

File: nodejs/week2/meal-sharing/src/backend/api/meals-router.js (hunk 3 above)
Comment (alina-kho) [reply nodejs/week2/meal-sharing/src/backend/api/meals-router.js:70]: fixed it :)

**Pull request merged**